modules/vba_documents/lib/vba_documents/deployment.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module VBADocuments
+  module Deployment
+    class << self
+      send(:attr_accessor, :environment)
+    end
+
+    def self.fetch_environment
+      prefix = Settings.vba_documents.location.prefix
+      prefix =~ /https:.*?-(dev|staging|sandbox|prod)-.*/
+      begin
+        Regexp.last_match(1).to_sym
+      rescue
+        :unknown_environment
+      end
```

Comment:
Could we avoid using rescuing for flow control? I think this will achieve the same thing.

```suggestion
      (Regexp.last_match(1) || :unknown_environment).to_sym
```